### PR TITLE
MINOR: add missing TimeoutException to Producer.send JavaDocs

### PR DESCRIPTION
**Pull request by mjsax**
Call for review @guozhangwang @abbccdda @hachikuji

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 1]
```diff
@@ -874,6 +874,7 @@ public void abortTransaction() throws ProducerFencedException {
      *                               when send is invoked after producer has been closed.
      * @throws InterruptException If the thread is interrupted while blocked
      * @throws SerializationException If the key or value are not valid objects given the configured serializers
+     * @throws TimeoutException If the record could not be appended to the send buffer within {@code max.block.ms}.
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: `..the sender buffer due to memory unavailable..`

**Review by guozhangwang — APPROVED** [review]
LGTM, just adding it is due to memory pressure.

**Review by abbccdda — APPROVED** [review]
LGTM!

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 2]
```diff
@@ -874,6 +874,7 @@ public void abortTransaction() throws ProducerFencedException {
      *                               when send is invoked after producer has been closed.
      * @throws InterruptException If the thread is interrupted while blocked
      * @throws SerializationException If the key or value are not valid objects given the configured serializers
+     * @throws TimeoutException If the record could not be appended to the send buffer due to memory unavailable within {@code max.block.ms}.
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: There is another possible reason. "metadata could not be refreshed within {@code max.block.ms}"

**Review by chia7712 — APPROVED** [review]
LGTM

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java (hunk 2 above)
Comment (guozhangwang) [reply clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: Yes, we should add this as a possible reason as well.

**Pull request merged**